atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -37,41 +37,124 @@
 
 /**
  * Provides the methods for a transaction with the key-value store.
+ *
+ * In general: users may assume that if maps (including sorted maps) keyed on byte[] are returned to the user,
+ * they may be accessed, including via random access, via any byte array that is equivalent in terms of
+ * {@link java.util.Arrays#equals(byte[], byte[])}.
+ *
  * @see TransactionManager
  */
 public interface Transaction {
 
+    /**
+     * Returns a mapping of rows to {@link RowResult}s within {@code tableRef} for the specified {@code rows}, loading
+     * columns according to the provided {@link ColumnSelection}.
+     *
+     * The returned {@link SortedMap} is sorted on the byte order of row keys; the ordering of the input parameter
+     * {@code rows} is irrelevant.
+     *
+     * If there are rows with no cells matching the provided {@link ColumnSelection}, they will not be present in the
+     * {@link Map#keySet()} of the output map at all. This accounts for local writes and deletes: a row written
+     * to locally (even if not persisted) will be present, and a row which is completely deleted locally (even if
+     * existing in the key value service) will be absent.
+     *
+     * @param tableRef table to load rows from
+     * @param rows rows to be loaded
+     * @param columnSelection columns to load from the given rows
+     * @return a mapping of rows to the columns matching the provided column selection
+     */
     @Idempotent
     SortedMap<byte[], RowResult<byte[]>> getRows(
             TableReference tableRef,
             Iterable<byte[]> rows,
             ColumnSelection columnSelection);
 
+    /**
+     * Returns a mapping of rows to {@link BatchingVisitable}s (which may be thought of as iterators over columns)
```

Comment:
nit: "which may be thought of as iterators over columns". Unclear of the value of putting this here? you are linking to the interface, perhaps the interface should have this sentence?

Reply:
Just an idea, so can be noop: perhaps you don't actually want to use "BatchingVisitable" of "BatchColumnRangeSelection" and explain it thus:

"Returns a mapping of requested rows to corresponding columns in this table. Only columns matching the single column range selection will be returned." @Jolyon-S  can provide better english.

Maybe this reads worse, hard to say.

Reply:
I think changing this to the version without the classes seems reasonable.